src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -2476,6 +2468,372 @@ public static void DCJS_RoundtrippingDateTime()
+
+        var baseline = $"\"\\/Date(1291161600000{timeZoneString})\\/\"";
+        if (Environment.OSVersion.VersionString.ToLower().Contains("osx") || Environment.OSVersion.VersionString.ToLower().Contains("windows"))
+        {
+            baseline = $"\"\\/Date(1291190400000{timeZoneString})\\/\"";
+        }
+        var actual = SerializeAndDeserialize(value, baseline, dcjsSettings);
+        Assert.Equal(value, actual);
+
+        var DTF_class = new JsonTypes.DTF_class()
+        {
+            dt1 = new DateTime(1, 1, 1, 3, 58, 32),
+            dt2 = new DateTime(2010, 12, 20),
+            dt3 = new DateTime(1998, 1, 1),
+            dt4 = new DateTime(1, 1, 1, 3, 58, 32, DateTimeKind.Utc)
+        };
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_yyyygg };
+        var actual2 = SerializeAndDeserialize(DTF_class, "{\"dt1\":\"0001 A.D.\",\"dt2\":\"2010 A.D.\",\"dt3\":\"1998 A.D.\",\"dt4\":\"0001 A.D.\"}", dcjsSettings);
+        Assert.NotNull(actual2);
+        Assert.True(actual2.dt1 == new DateTime(1, 1, 1));
+        Assert.True(actual2.dt2 == new DateTime(2010, 1, 1));
+        Assert.True(actual2.dt3 == new DateTime(1998, 1, 1));
+        Assert.True(actual2.dt4 == new DateTime(1, 1, 1));
+
+        var graph = new DateTimeOffset(2008, 5, 1, 8, 6, 32, new TimeSpan(1, 0, 0));
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_DMMMM };
+        var actual3 = SerializeAndDeserialize(graph, "{\"DateTime\":\"1, mayo\",\"OffsetMinutes\":60}", dcjsSettings);
+        Assert.NotNull(actual3);
+        Assert.True(actual3 == new DateTimeOffset(2017, 5, 1, 1, 0, 0, new TimeSpan(1, 0, 0)));
+
+        var dt35832 = new DateTime(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day, 3, 58, 32);
+        var dt = new DateTime(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day);
+        var expected_DT_list = "[\"03:58:32.00 a.m.\",\"12:00:00.00 a.m.\",\"12:00:00.00 a.m.\",\"03:58:32.00 a.m.\"]";
+        if (Environment.OSVersion.VersionString.ToLower().Contains("ubuntu"))
+        {
+            expected_DT_list = "[\"03:58:32.00 a. m.\",\"12:00:00.00 a. m.\",\"12:00:00.00 a. m.\",\"03:58:32.00 a. m.\"]";
+        }
+        dcjsSettings = new DataContractJsonSerializerSettings() { DateTimeFormat = jsonTypes.DTF_hmsFt };
+        var actual4 = SerializeAndDeserialize(jsonTypes.DT_List, expected_DT_list, dcjsSettings);
+        Assert.NotNull(actual4);
+        Assert.True(actual4[0] == dt35832);
+        Assert.True(actual4[1] == dt);
+        Assert.True(actual4[2] == dt);
+        Assert.True(actual4[3] == dt35832);
+        
+        var expected_DT_dictionary = "[{\"Key\":\"03:58:32.00 a.m.\",\"Value\":\"03:58:32.00 a.m.\"},{\"Key\":\"12:00:00.00 a.m.\",\"Value\":\"12:00:00.00 a.m.\"}]";
+        if (Environment.OSVersion.VersionString.ToLower().Contains("ubuntu"))
```

Comment:
Can you please open an issue for tracking this behavior difference and mark the test as ActiveIssue?